token/js/client/token.js

```diff
@@ -2103,4 +2146,45 @@ export class Token {
       data,
     });
   }
+
+  static async getAssociatedTokenAddress(
+    associatedProgramId: PublicKey,
+    programId: PublicKey,
+    owner: PublicKey,
+    mint: PublicKey,
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [owner.toBuffer(), programId.toBuffer(), mint.toBuffer()],
+        associatedProgramId,
+      )
+    )[0];
+  }
+
+  static createAssociatedTokenAccountInstruction(
+    associatedProgramId: PublicKey,
+    programId: PublicKey,
+    payer: PublicKey,
+    owner: PublicKey,
+    mint: PublicKey,
+    associatedAccount: PublicKey,
+  ): TransactionInstruction {
+    const data = Buffer.alloc(0);
```

Comment:
maybe, is there more better way to create an empty Buffer?

Reply:
I've seen this done elsewhere, seems fine to me